exporter/awskinesisexporter/internal/batch/aggregator.go

```diff
@@ -0,0 +1,75 @@
+package batch
+
+import (
+	"crypto/md5"
+
+	k "github.com/aws/aws-sdk-go/service/kinesis"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	magicNumber           = []byte{0xF3, 0x89, 0x9A, 0xC2}
+	partitionKeyIndexSize = 8
+	maxAggregationCount   = 4294967295
+	maxRecordSize         = 1 << 20
+)
+
+type Aggregator struct {
+	buf    []*Record
+	pkeys  []string
+	nbytes int
+}
+
+func (a *Aggregator) IsRecordAggregative(data []byte, partitionKey string) bool {
+	return data != nil && len(data)+len([]byte(partitionKey)) <= maxRecordSize
+}
+
+func (a *Aggregator) IsBatchFull(data []byte, partitionKey string) bool {
```

Comment:
This method name needs to change to `CheckIfFull() (remaining int, full bool)` or something to that affect since `IsBatchFull` would infer to me that I am checking at its current state if there is space remaining, not check if there is space remaining while also trying to add in a new record / data.